.github/workflows/ci.yml

```diff
@@ -54,42 +99,73 @@ jobs:
     steps:
       - uses: actions/checkout@v2
         name: Checkout code
+
+      - name: Pull snuba CI images
+        run: |
+          docker pull ghcr.io/getsentry/snuba-ci:${{ github.sha }} || \
+            docker pull ghcr.io/getsentry/snuba-ci:${{ needs.snuba-image.outputs.branch }} || \
+            docker pull ghcr.io/getsentry/snuba-ci:latest || true
+
+      - name: Build snuba docker image for CI
+        run: |
+          docker build . \
+            --build-arg PYTHON_VERSION=3.8 \
```

Comment:
I think you need to use `--build-arg BUILDKIT_INLINE_CACHE=1` and the env `DOCKER_BUILDKIT=1`